13_07_2017/3D_slab_modes.py

```diff
@@ -0,0 +1,779 @@
+vis_modules_strings = ['show_density', 'show_density_pert', 'show_mag', 'show_mag_scale', 
+                       'show_mag_fade', 'show_mag_vec', 'show_vel_front', 'show_vel_front_pert',
+                       'show_vel_top', 'show_vel_top_pert', 'show_disp_top', 'show_disp_front']
+vis_mod_string = ''
+for i in range(len(vis_modules)):
+    if vis_modules[i] == True:
+        vis_mod_string = vis_mod_string + vis_modules_strings[i][5:] + '_'
+
+
+# Set to True if you would like the dispersion diagram with chosen mode highlighted.
+show_dispersion = False
+#show_dispersion = True
+
+# Wanna see the animation? Of course you do
+#show_animation = False
+show_animation = True
+
+# Basic plot to see which eigensolutions have been found.
+show_quick_plot = False
+#show_quick_plot = True
+
+# Video resolution
+#res = (1920,1080) # There is a problem with this resolution- height must be odd number - Mayavi bug apparently
+res = tuple(101 * np.array((16,9)))
+#res = tuple(51 * np.array((16,9)))
+#res = tuple(21 * np.array((16,9)))
+
+number_of_frames = 50
+
+# Frames per second of output video
+fps = 20
+
+#save_images = False
+save_images = True
+
+#make_video = False
+make_video = True
+
+#
+##
+###
+####
+#####
+######
+#######
+########
+#########
+
+# Variable definitions:
+# x = k*x
+# y = k*y
+# z = k*z
+# W = omega/k
+# K = k*x_0
+# t = omega*t
+
+if np.array([show_density, show_vel_front_pert, show_vel_top_pert]).any() == True and mode in alfven_mode_options:
+    raise NameError('Cannot show density or vel/disp pert for this mode')
+
+# Loop through selected modes
+for mode_ind in [0,1]:#range(8,14): # for all others. REMEMBER SBB pparameters
+#for mode_ind in [14,15]: #for fast body surf. REMEMBER SBS parameters
+#for mode_ind in [16, 17]:
+#for mode_ind in [13]: #for an individual mode
+#for mode_ind in range(2,14): 
+    if mode_ind not in range(len(mode_options)):
+        raise NameError('Mode not in mode_options')
+        
+    # (note that fast surface modes, i.e. 14 and 15, can only be 
+    # found with SBS parameters in slab_functions...)
+    mode = mode_options[mode_ind]    
+    
+    # Specify oscillation parameters
+    if mode in slow_surf_mode_options + alfven_mode_options:
+        K = 2.
+    elif mode in slow_body_mode_options:
+        K = 8.
+    elif mode in fast_body_1_mode_options:
+        K = 8.
+    elif mode in fast_body_2_mode_options:
+        K = 15.
+    elif mode in fast_body_3_mode_options:
+        K = 22.
+    elif mode in fast_surf_mode_options:
+        K = 8.
+    else:
+        raise NameError('Mode not found')
+
+# Specify density ratio R1 := rho_1 / rho_0
+#    R1 = 1.5 # Higher denisty on left than right
+#    R1 = 1.8
+#    R1 = 1.9 # Disp_diagram will only work for R1=1.5, 1.8, 2.0       
+    R1 = 2. # Symmetric slab
+    
+    # Reduce number of variables in dispersion relation
+    def disp_rel_asym_2var(W, K):
+        return sf.disp_rel_asym(W, K, R1)
```

Comment:
Indeed, you can absolutely do that. Can't you just pass in R1 as an argument though? If it's the same for every loop then making it a partial function makes sense in principle but for such a simple thing it may be over-engineering it a bit.